### PR TITLE
refactor: improve plugin types, modernize monorepo structure

### PR DESCRIPTION
**Pull request by ssalbdivad**
This PR includes:

- improvements to type-safety and structure for url-loader
- update monorepo infrastructure (prettier, eslint, add tsconfig paths)
- apply updated formatting and lint rules
- rewrite createCloudinaryUrl types to remove zod dependency in build output

### CODE REVIEW

File: packages/types/src/types/cloudinary-video-player.ts
Comment (Baroshem) [inline packages/types/src/types/cloudinary-video-player.ts:31]: nitpick: I think you can remove the comment here because it is no longer up to date

File: packages/url-loader/src/plugins/cropping.ts
Comment (Baroshem) [inline packages/url-loader/src/plugins/cropping.ts:175]: question: does this work? I mean, are you able to modify the height and gravity variables if they are changed to consts?

File: packages/util/package.json
Comment (Baroshem) [inline packages/util/package.json:29]: question: should all these packages be removed? Like `vitest` for example for testing?

File: packages/url-loader/src/types/video.ts
Comment (Baroshem) [inline packages/url-loader/src/types/video.ts:15]: question: I am not sure if this change will actually fix the issue on the vue projects. I think this would need to be tested by me. Could you let me know once you will release a `rc` version or beta of this new approach so that I can test it out and let you know if it makes a difference?

Comment (Baroshem) [reply packages/url-loader/src/types/video.ts:15]: Because eventually it is `type _VideoOptions = z.infer<typeof videoOptionsSchema>;`

**Review by Baroshem — COMMENTED** [review]
Hey there!

A lot of great work. I have added a few questions and comments :)

File: packages/url-loader/src/plugins/cropping.ts
Comment (ssalbdivad) [reply packages/url-loader/src/plugins/cropping.ts:175]: This was actually based on an eslint rule I enabled to catch cases where `let` is not needed because the variable is never reassigned:

https://eslint.org/docs/latest/rules/prefer-const

File: packages/util/package.json
Comment (ssalbdivad) [reply packages/util/package.json:29]: These weren't removed, I just moved them to `devDependencies` at the top-level of the monorepo.

That way, the versions aren't maintained in multiple places and since they're only used in development across packages, it's generally easier to maintain dev deps this way in a monorepo.

File: packages/url-loader/src/types/video.ts
Comment (ssalbdivad) [reply packages/url-loader/src/types/video.ts:15]: Yes, this was not the change designed to address the resolution issue with `zod`.

Take a look at `packages/url-loader/build.ts`. I only applied the strategy to `ConstructUrlProps` the primary parameter of `constructCloudinaryUrl`. However, this pattern could easily be extended to compile any other zod-inferred types down to standard TS.

File: packages/types/src/types/cloudinary-video-player.ts
Comment (ssalbdivad) [reply packages/types/src/types/cloudinary-video-player.ts]: Good catch! Fixed

File: packages/url-loader/src/plugins/cropping.ts
Comment (Baroshem) [reply packages/url-loader/src/plugins/cropping.ts:175]: Ahh sorry, I just see now that gravity is still a let. Closing the comment!

File: packages/util/package.json
Comment (Baroshem) [reply packages/util/package.json:29]: Very nice improvement!

File: packages/url-loader/src/types/video.ts
Comment (Baroshem) [reply packages/url-loader/src/types/video.ts:15]: I see, thanks for clarification :)

File: packages/types/package.json
Comment (ghostdevv) [inline packages/types/package.json:12]: we should cover bases if building seperate esm and cjs builds but is probably not necessary to change this package at all since it's only types

```suggestion
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "require": {
        "types": "./dist/index.d.cts",
        "default": "./dist/index.cjs"
      },
      "import": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
```

File: packages/url-loader/package.json
Comment (ghostdevv) [inline packages/url-loader/package.json:37]: this should be a dev dep

File: packages/util/package.json
Comment (ghostdevv) [inline packages/util/package.json:12]: Similar exports thing here

File: tsconfig.json
Comment (ghostdevv) [inline tsconfig.json:23]: I haven't seen this needed before - is it used for anything?

**Review by ghostdevv — CHANGES_REQUESTED** [review]
We're missing a prettier config too something like:

`.prettierrc`
```json
 {
    "singleQuote": true,
    "quoteProps": "as-needed",
    "trailingComma": "all",
    "bracketSpacing": true,
    "arrowParens": "always",
    "semi": true,
    "useTabs": true,
    "tabWidth": 4,
}
```

(you might want to change `singleQuote` to false if that's what you want to use)

File: packages/types/package.json
Comment (ssalbdivad) [reply packages/types/package.json:12]: I agree this could be simplified but I went with reusing the same build process we have across packages.

I hadn't seen this `.d.cts` setup either but it seems to be what the latest version of tsup is doing by default? If you'd like to follow up with a future PR adjusting this I think it could be improved but there are already quite a lot of changes in this PR and unfortunately I don't have additional hours to spend updating it at this point.

For now, I'm not sure if the `.d.cts` files would be resolved for `.cjs` imports or if it would just resolve to the `.d.ts` files but in any case the types should be the same in the vast majority of circumstances so I don't think it should break anything.

File: tsconfig.json
Comment (ssalbdivad) [reply tsconfig.json:23]: It is not needed but it ensures types across packages stay in sync during dev without having to rebuild. It also works for runtime code using a loader like `tsx` that respects tsconfig paths.

See https://colinhacks.com/essays/live-types-typescript-monorepo for more info

File: .eslintrc.cjs
Comment (ssalbdivad) [reply .eslintrc.cjs:1]: It's kind of complicated. You can upgrade to flat configs with eslint 9+ but many plugins including some of those that the repo was using don't support that yet so it would be a larger migration.

I just tried to keep everything as consistent as possible while migrating to a newer version.

**Comment by ssalbdivad** [comment]
@ghostdevv In terms of prettier config, you can add one if you'd like. For now it is just using default settings.

File: packages/url-loader/package.json
Comment (ssalbdivad) [reply packages/url-loader/package.json]: Whoops, my bad! Just pushed a commit fixing this.

**Comment by ssalbdivad** [comment]
@ghostdevv @Baroshem Is there anything else I can do to help get this merged?

If there are any regressions I do want to ensure those are addressed. I am under the impression the remaining feedback could be addressed in a future PR and will not cause any issues in the meantime, but please let me know if I am wrong about that.

I want to make sure this PR doesn't languish too long given the potential for merge conflicts!

**Comment by colbyfayock** [comment]
I think I'm the blocker. I'd like to take a look at this before merging it
in but there won't be anything else that gets merged in before it. I'll do
what I can to take a look soon with my travel schedule.

**Comment by Baroshem** [comment]
No blockers from my side. Waiting for the release to test it in Nuxt Cloudinary :) 

Idea: Maybe we could use https://github.com/stackblitz-labs/pkg.pr.new to test it even faster? @colbyfayock @ssalbdivad @ghostdevv

**Comment by colbyfayock** [comment]
thanks @ssalbdivad! push some fixes to the tests from the restructing, but going to get this merged into a beta release so we can start testing it

**Comment by colbyfayock** [comment]
also, i dont believe this has any breaking changes, correct?

**Comment by colbyfayock** [comment]
@all-contributors please add @ssalbdivad for code

**Comment by ssalbdivad** [comment]
@colbyfayock Yes there should not be any breaking changes!

**Pull request merged**
